### PR TITLE
No default values in BacnetObjectType

### PR DESCRIPTION
**Pull request by JoelRedstone**
Hi @grafnu,
I have made a small change that allow to decide whether or not to include default values when creating a bacnetObjectType in a slave device.
By default the library include all mandatory property when declaring a new bacnetObjectType so
I need this change to make a faux device without mandatory properties (like "present value", ecc..) to make the bacnet.pics test fail.
PTAL

### CODE REVIEW

File: src/main/java/com/serotonin/bacnet4j/obj/BACnetObject.java [hunk 1]
```diff
@@ -76,14 +76,27 @@
     private final ObjectIdentifier id;
     private final Map<PropertyIdentifier, Encodable> properties = new HashMap<PropertyIdentifier, Encodable>();
     private final List<ObjectCovSubscription> covSubscriptions = new ArrayList<ObjectCovSubscription>();
+    private boolean setDefaultValues = true;
 
     public BACnetObject(LocalDevice localDevice, ObjectIdentifier id) {
         this.localDevice = localDevice;
+        if (id == null)
```
Comment (grafnu) [inline src/main/java/com/serotonin/bacnet4j/obj/BACnetObject.java]: Typically always put braces even if one line, so 

if (id == null) {
  throw ...
}

File: src/main/java/com/serotonin/bacnet4j/obj/BACnetObject.java [hunk 2]
```diff
@@ -76,14 +76,27 @@
...
+        setDefaultValues();
+    }
 
+    public BACnetObject(LocalDevice localDevice, ObjectIdentifier id, boolean setDefaultValues) throws BACnetServiceException {
+        this.localDevice = localDevice;
+        this.setDefaultValues = setDefaultValues;
         if (id == null)
             throw new IllegalArgumentException("object id cannot be null");
         this.id = id;
+        if(setDefaultValues){ setDefaultValues(); }
```
Comment (grafnu) [inline src/main/java/com/serotonin/bacnet4j/obj/BACnetObject.java]: Don't collapse to one line, keep braces and make 3 lines.

File: src/main/java/com/serotonin/bacnet4j/obj/BACnetObject.java [hunk 3]
```diff
@@ -76,14 +76,27 @@
...
+        else { setProperty(PropertyIdentifier.objectName, new CharacterString(id.toString())); }
```
Comment (grafnu) [inline src/main/java/com/serotonin/bacnet4j/obj/BACnetObject.java]: ditto

**Review by grafnu — CHANGES_REQUESTED** [review]
Just a few minor style comments.

**Comment by JoelRedstone** [comment]
@grafnu 
PTAL

**Review by grafnu — APPROVED** [review]

**Comment by grafnu** [comment]
Squashed and merged.  I made one small change -- instead of something like

if (a){ 

it should be

if (a) {

(missing space)... just a small style nit!

**Pull request merged**